File: count_odd_num_interval_range.cpp
// 1523. Count Odd Numbers in an Interval Range
#include "leetcode.h"

// given two non-negative integers 'low' and 'high' return the
// count of odd numbers  * between 'low' and 'high' (incusively)

class Solution {
public:
  int countOdds(int low, int high) {
    int ans = 0;
    if (low % 2) {
      ans += (high - low) / 2;
      ans++;
    } else
      ans += ceil(double(high - low) / 2.0);
    return ans;
  }
};

int main() {
  Solution obj;
  cout << obj.countOdds(3, 7) << endl;  // expect: 3
  cout << obj.countOdds(8, 10) << endl; // expect: 1
}

File: count_odd_num_interval_range.py
# 1523. Count Odd Numbers in an Interval Range

"""
given two non-negative integers 'low' and 'high' return the
count of odd numbers  * between 'low' and 'high' (incusively)
"""


class Solution(object):
    def countOdds(self, low, high):
        """
        :type low: int
        :type high: int
        :rtype: int
        """
        return (high + 1) // 2 - low // 2


if __name__ == "__main__":
    obj = Solution()
    print(obj.countOdds(low=3, high=7))
    print(obj.countOdds(low=8, high=10))
